src/main/java/org/jenkinsci/plugins/docker/commons/credentials/ImageNameValidator.java

```diff
@@ -129,6 +147,43 @@ public static void checkUserAndRepo(@NonNull String userAndRepo) throws FormVali
         }
     }
 
+    /**
+     * A digest starts with 'sha256:' and must be valid ASCII and may contain
+     * lowercase and digits.
+     * A digest contains 71 additional characters.
+     *
+     * @see <a href=
+     *      "https://docs.docker.com/engine/reference/commandline/images/#list-the-full-length-image-ids">docker
+     *      digests</a>
+     */
+    public static final Pattern VALID_DIGEST = Pattern.compile("^sha256:([a-z0-9]){64}$");
```

Comment:
Should we also allow other digest forms supported by the [OCI Image spec](https://github.com/opencontainers/image-spec/blob/v1.0.1/descriptor.md#digests)?

```
digest                ::= algorithm ":" encoded
algorithm             ::= algorithm-component (algorithm-separator algorithm-component)*
algorithm-component   ::= [a-z0-9]+
algorithm-separator   ::= [+._-]
encoded               ::= [a-zA-Z0-9=_-]+
```

`sha256` and `sha512` are registered and as-such have [further validation rules](https://github.com/opencontainers/image-spec/blob/v1.0.1/descriptor.md#registered-algorithms)

It might be better to use either that link, or https://docs.docker.com/registry/spec/api/#content-digests (the predecessor to [the OCI Distribution spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#definitions) which uses the OCI Image spec digest rules I linked above), since the current links only shows _examples_ of valid digests but does not define the validation rule.

The Docker Registry V2's rule accepts more than the OCI sha256 rule, and is neither a strict subset nor superset of the OCI general algorithm rule, so I think we can't expect to get a validation rule that doesn't still carry the risk of being rejected by the registry being talked to.

Reply:
OK, which one to implement? I like to get this done because of the currently breaking issue? Probably the OCI spec ist best.